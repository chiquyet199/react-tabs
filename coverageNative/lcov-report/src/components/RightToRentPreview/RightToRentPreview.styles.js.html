<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/RightToRentPreview/RightToRentPreview.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/RightToRentPreview</a> RightToRentPreview.styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/7</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from "styled-components/native";
&nbsp;
export const Container = styled.View``;
&nbsp;
export const StyledText = styled.Text`
  font-size: 16;
  margin-top: 8;
  font-style: ${<span class="fstat-no" title="function not covered" >({</span>italic}) =&gt; (<span class="cstat-no" title="statement not covered" >italic ? "italic" : "normal")</span>}
  color: ${<span class="fstat-no" title="function not covered" >({</span>alert, theme}) =&gt; (<span class="cstat-no" title="statement not covered" >alert ? theme.errorColor : theme.textColor)</span>};
`;
&nbsp;
export const StyledImage = styled.Image`
  height: 342px;
  width: 340px;
`;
&nbsp;
export const ScanningMetadata = styled.Text`
  font-size: 12;
  color: ${<span class="fstat-no" title="function not covered" >({</span>theme}) =&gt; <span class="cstat-no" title="statement not covered" >theme.textColor}</span>;
`;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:50 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
