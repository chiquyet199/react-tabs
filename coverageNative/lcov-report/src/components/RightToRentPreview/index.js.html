<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/RightToRentPreview/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/RightToRentPreview</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.31% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/12</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, {Component} from "react";
import {themed} from "../../utils/common";
import {connect} from "react-redux";
import {
  Container,
  StyledText,
  ScanningMetadata,
} from "./RightToRentPreview.styles";
import {arrayOf, func, oneOf, shape, string} from "prop-types";
import moment from "moment";
import {fetchUserById} from "../../actions/users";
&nbsp;
export class RightToRentPreview extends Component {
  componentDidMount() {
    const {
      getUser,
      passport: {
        data: {seenBy},
      },
    } = this.props;
    getUser(seenBy);
  }
&nbsp;
  getScanningMetadata() {
    const {
      passport: {
        data: {seenAt},
        translations: {seenBy},
      },
      scannedByName,
    } = this.props;
&nbsp;
    return `${seenBy}${scannedByName}, ${moment(seenAt).format("DD.MM.YYYY")}`;
  }
&nbsp;
  render() {
    const {passport, theme, globaltheme, scannedByName} = this.props;
    const hasRightToRent = passport.data.documents.length &gt; 0;
    const {
      translations,
      data: {documentStatus: status},
    } = passport;
&nbsp;
    const textProps = {
      italic: status === "PENDING",
      alert: status === "FAIL",
      theme: theme[globaltheme],
    };
&nbsp;
    return (
      &lt;Container&gt;
        &lt;StyledText {...textProps}&gt;
          {hasRightToRent
            ? `${translations.rightToRent}${translations[status.toLowerCase()]}`
            : translations.uploadRight}
        &lt;/StyledText&gt;
        {scannedByName &amp;&amp; (
          &lt;ScanningMetadata theme={theme[globaltheme]}&gt;
            {this.getScanningMetadata()}
          &lt;/ScanningMetadata&gt;
        )}
      &lt;/Container&gt;
    );
  }
}
&nbsp;
RightToRentPreview.propTypes = {
  passport: shape({
    data: shape({
      id: string,
      provider: oneOf(["TRUST_ID"]).isRequired,
      tenantId: string,
      documentStatus: oneOf(["PENDING", "SUBMITTED", "PASS", "FAIL"])
        .isRequired,
      seenBy: string.isRequired,
      seenAt: string.isRequired,
      status: oneOf(["GREEN", "YELLOW", "RED", "NONE"]).isRequired,
      documents: arrayOf(
        shape({
          id: string,
          type: oneOf(["PASSPORT", "BIOMETRIC_RESIDENCE_PERMIT"]),
          images: arrayOf(string).isRequired, // document refs
          expires: string,
        }),
      ).isRequired,
    }).isRequired,
    scannedByName: string,
    translations: shape({
      rightToRent: string.isRequired,
      uploadRight: string.isRequired,
      pending: string.isRequired,
      submitted: string.isRequired,
      pass: string.isRequired,
      fail: string.isRequired,
      seenBy: string.isRequired,
    }).isRequired,
  }).isRequired,
  theme: shape({
    textColor: string,
    errorColor: string,
  }).isRequired,
  getUser: func.isRequired,
  scannedByName: string,
  globaltheme: oneOf(["dark", "light"]),
};
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >({</span>state: profile}) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  scannedByName: profile &amp;&amp; `${profile.firstName} ${profile.lastName}`,
});
&nbsp;
const mapDispatchToProps = {
  getUser: fetchUserById,
};
&nbsp;
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(themed(RightToRentPreview));
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:50 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
