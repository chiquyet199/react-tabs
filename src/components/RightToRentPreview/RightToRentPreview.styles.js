import styled from "styled-components/native";

export const Container = styled.View``;

export const StyledText = styled.Text`
  font-size: 16;
  margin-top: 8;
  font-style: ${({italic}) => (italic ? "italic" : "normal")}
  color: ${({alert, theme}) => (alert ? theme.errorColor : theme.textColor)};
`;

export const StyledImage = styled.Image`
  height: 342px;
  width: 340px;
`;

export const ScanningMetadata = styled.Text`
  font-size: 12;
  color: ${({theme}) => theme.textColor};
`;
