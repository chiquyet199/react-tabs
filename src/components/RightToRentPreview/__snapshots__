/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/RightToRentPreview/__native_snapshots__