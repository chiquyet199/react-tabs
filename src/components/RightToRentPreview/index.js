import React, {Component} from "react";
import {themed} from "../../utils/common";
import {connect} from "react-redux";
import {
  Container,
  StyledText,
  ScanningMetadata,
} from "./RightToRentPreview.styles";
import {arrayOf, func, oneOf, shape, string} from "prop-types";
import moment from "moment";
import {fetchUserById} from "../../actions/users";

export class RightToRentPreview extends Component {
  componentDidMount() {
    const {
      getUser,
      passport: {
        data: {seenBy},
      },
    } = this.props;
    getUser(seenBy);
  }

  getScanningMetadata() {
    const {
      passport: {
        data: {seenAt},
        translations: {seenBy},
      },
      scannedByName,
    } = this.props;

    return `${seenBy}${scannedByName}, ${moment(seenAt).format("DD.MM.YYYY")}`;
  }

  render() {
    const {passport, theme, globaltheme, scannedByName} = this.props;
    const hasRightToRent = passport.data.documents.length > 0;
    const {
      translations,
      data: {documentStatus: status},
    } = passport;

    const textProps = {
      italic: status === "PENDING",
      alert: status === "FAIL",
      theme: theme[globaltheme],
    };

    return (
      <Container>
        <StyledText {...textProps}>
          {hasRightToRent
            ? `${translations.rightToRent}${translations[status.toLowerCase()]}`
            : translations.uploadRight}
        </StyledText>
        {scannedByName && (
          <ScanningMetadata theme={theme[globaltheme]}>
            {this.getScanningMetadata()}
          </ScanningMetadata>
        )}
      </Container>
    );
  }
}

RightToRentPreview.propTypes = {
  passport: shape({
    data: shape({
      id: string,
      provider: oneOf(["TRUST_ID"]).isRequired,
      tenantId: string,
      documentStatus: oneOf(["PENDING", "SUBMITTED", "PASS", "FAIL"])
        .isRequired,
      seenBy: string.isRequired,
      seenAt: string.isRequired,
      status: oneOf(["GREEN", "YELLOW", "RED", "NONE"]).isRequired,
      documents: arrayOf(
        shape({
          id: string,
          type: oneOf(["PASSPORT", "BIOMETRIC_RESIDENCE_PERMIT"]),
          images: arrayOf(string).isRequired, // document refs
          expires: string,
        }),
      ).isRequired,
    }).isRequired,
    scannedByName: string,
    translations: shape({
      rightToRent: string.isRequired,
      uploadRight: string.isRequired,
      pending: string.isRequired,
      submitted: string.isRequired,
      pass: string.isRequired,
      fail: string.isRequired,
      seenBy: string.isRequired,
    }).isRequired,
  }).isRequired,
  theme: shape({
    textColor: string,
    errorColor: string,
  }).isRequired,
  getUser: func.isRequired,
  scannedByName: string,
  globaltheme: oneOf(["dark", "light"]),
};

const mapStateToProps = ({state: profile}) => ({
  scannedByName: profile && `${profile.firstName} ${profile.lastName}`,
});

const mapDispatchToProps = {
  getUser: fetchUserById,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(themed(RightToRentPreview));
