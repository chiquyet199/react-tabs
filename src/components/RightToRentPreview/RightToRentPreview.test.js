import React from "react";
import {shallow} from "enzyme";
import {RightToRentPreview} from "../RightToRentPreview";
import {StyledText, ScanningMetadata} from "./RightToRentPreview.styles";
import {extend} from "underscore";

describe("RightToRentPreview Tests", () => {
  describe("Snapshot", () => {
    it("matches snapshot", () => {
      expect(
        getComponent({passport: passportWithPhoto("PENDING")}),
      ).toMatchSnapshot();
    });

    it("renders correct status", () => {
      const statuses = ["PENDING", "SUBMITTED", "PASS", "FAIL"];
      const expectedText = [
        "Right to rent pending",
        "Right to rent submitted",
        "Right to rent verified",
        "Right to rent declined",
      ];

      for (let x = 0; x < statuses.length; x += 1) {
        const status = statuses[x];
        expect(
          getComponent({passport: passportWithPhoto(status)}).find(StyledText),
        ).toHaveProp("children", expectedText[x]);
      }
    });

    it("passes italic prop to StyledText when status is PENDING", () => {
      expect(
        getComponent({passport: passportWithPhoto("PENDING", "YELLOW")}).find(
          StyledText,
        ),
      ).toHaveProp("italic", true);
    });

    it("passes alert prop to StyledText when status is FAIL", () => {
      expect(
        getComponent({passport: passportWithPhoto("FAIL", "RED")}).find(
          StyledText,
        ),
      ).toHaveProp("alert", true);
    });

    it("Correctly renders photoUpload text", () => {
      expect(
        getComponent({
          passport: passportWithoutPhoto("PENDING", "YELLOW"),
        }).find(StyledText),
      ).toHaveProp("children", "Upload a photo");
    });

    it("requests name of seenBy id on mount", () => {
      const passport = passportWithPhoto("PENDING", "YELLOW");
      passport.data.seenBy = "abc-123-def-456";
      const getUser = jest.fn();
      getComponent({passport, getUser});
      expect(getUser).toHaveBeenCalledWith(passport.data.seenBy);
    });

    it("adds scanned by metadata when scannedByName prop is present", () => {
      const passport = passportWithPhoto("PENDING");
      const component = getComponent({passport});
      expect(component.find(ScanningMetadata)).not.toExist();
      component.setProps({
        scannedByName: "Becky Hammond",
      });
      component.update();
      expect(component.find(ScanningMetadata)).toHaveProp(
        "children",
        "Seen by Becky Hammond, 01.11.2018",
      );
    });
  });

  const getComponent = props => {
    const parsedProps = extend(
      {
        getUser: () => {},
        globaltheme: "light",
        passport: passportWithPhoto("PASS"),
        theme: {
          textColor: "black",
          errorColor: "orange",
        },
      },
      props,
    );

    return shallow(<RightToRentPreview {...parsedProps} />);
  };
});

const passportWithPhoto = (documentStatus, status = "GREEN") => ({
  data: {
    id: "abc-123",
    provider: "TRUST_ID",
    tenantId: "def-456",
    documentStatus,
    seenBy: "ghi-789",
    seenAt: "2018-11-01T12:37:00",
    status,
    documents: [
      {
        id: "mno-345",
        type: "PASSPORT",
        images: ["jkl-012"],
        expires: "2020-07-16T00:00:00",
      },
    ],
    messages: [],
  },
  translations: {
    rightToRent: "Right to rent ",
    uploadRight: "uploadRight",
    pending: "pending",
    submitted: "submitted",
    pass: "verified",
    fail: "declined",
    seenBy: "Seen by ",
  },
});

const passportWithoutPhoto = (documentStatus, status = "GREEN") => ({
  data: {
    id: "abc-123",
    provider: "TRUST_ID",
    tenantId: "def-456",
    documentStatus,
    seenBy: "ghi-789",
    seenAt: "2018-11-01T12:37:00",
    status,
    documents: [],
    messages: [],
  },
  translations: {
    rightToRent: "rightToRent",
    uploadRight: "Upload a photo",
    pending: "pending",
    submitted: "submitted",
    pass: "verified",
    fail: "declined",
    seenBy: "seenBy",
  },
});
