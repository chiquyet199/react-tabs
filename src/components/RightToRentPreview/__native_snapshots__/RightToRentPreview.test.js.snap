// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightToRentPreview Tests Snapshot matches snapshot 1`] = `
<Styled(Component)>
  <Styled(Text)
    alert={false}
    italic={true}
  >
    Right to rent pending
  </Styled(Text)>
</Styled(Component)>
`;
